src/main/java/org/opensearch/knn/quantization/models/quantizationState/MultiBitScalarQuantizationState.java

```diff
@@ -124,4 +125,45 @@ public byte[] toByteArray() throws IOException {
     public static MultiBitScalarQuantizationState fromByteArray(final byte[] bytes) throws IOException {
         return (MultiBitScalarQuantizationState) QuantizationStateSerializer.deserialize(bytes, MultiBitScalarQuantizationState::new);
     }
+
+    /**
+     * Calculates and returns the number of bytes stored per vector after quantization.
+     *
+     * @return the number of bytes stored per vector.
+     */
+    @Override
+    public int getBytesPerVector() {
+        // Check if thresholds are null or have invalid structure
+        if (thresholds == null || thresholds.length == 0 || thresholds[0] == null) {
+            throw new IllegalStateException("Error in getBytesStoredPerVector: The thresholds array is not initialized.");
+        }
+
+        // Calculate the number of bytes required for multi-bit quantization
+        return thresholds.length * thresholds[0].length;
+    }
+
+    @Override
+    public int getDimensions() {
+        // For multi-bit quantization, the dimension for indexing is the number of rows * columns in the thresholds array.
+        // Where number of column reprensents Dimesion of Original vector and number of rows equals to number of bits
+        return thresholds.length * thresholds[0].length;
```

Comment:
why getDimension is not checking `thresholds are null or have invalid structure`